ansible_base/rbac/api/permissions.py

```diff
@@ -80,22 +88,28 @@ def get_required_object_permissions(self, method, model_cls, view=None):
         # Remove add permissions, since they are handled in has_permission
         return [p for p in perms if not is_add_perm(p)]
 
+    def has_object_permission_by_codename(self, request, obj, perms):
+        return all(request.user.has_obj_perm(obj, perm) for perm in perms)
+
+    def model_is_valid(self, model_cls):
+        return permission_registry.is_registered(model_cls)
+
     def has_object_permission(self, request, view, obj):
         if not isinstance(obj, Model):
             logger.info(f'Object-permission check called with non-object {type(obj)} for {request.method}')
             return True  # for the DRF browsable API, showing PATCH form field
 
         queryset = self._queryset(view)
         model_cls = queryset.model
-        if not permission_registry.is_registered(model_cls):
+        if not self.model_is_valid(model_cls):
             if request.user.is_superuser:
                 return True
             logger.warning(f'User {request.user.pk} denied {request.method} to {obj._meta.model_name}, not in DAB RBAC permission registry')
             raise Http404
 
         perms = self.get_required_object_permissions(request.method, model_cls, view=view)
 
-        if not all(request.user.has_obj_perm(obj, perm) for perm in perms):
+        if not self.has_object_permission_by_codename(request, obj, perms):
```

Comment:
I don't love this, it's tortured. I am inclined to adjust `has_obj_perm` to do the special-cases for users. But I'm okay enough with this for now.